firefox_ui_tests/functional/locationbar/test_favicon_in_autocomplete.py

```diff
@@ -0,0 +1,87 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import time
+
+from marionette_driver import Wait
+
+from firefox_ui_harness.decorators import skip_under_xvfb
+from firefox_ui_harness import FirefoxTestCase
+
+
+class TestFaviconInAutocomplete(FirefoxTestCase):
+
+    def setUp(self):
+        FirefoxTestCase.setUp(self)
+
+        # Location bar suggests "History"
+        self.prefs.set_pref('browser.urlbar.default.behavior', 1)
+
+        self.test_urls = [self.marionette.absolute_url('layout/mozilla.html')]
+
+        self.test_string = 'mozilla'
+
+        self.autocomplete_results = self.browser.navbar.locationbar.autocomplete_results
+
+        self.places.remove_all_history()
+
+    def tearDown(self):
+        try:
+            self.autocomplete_results.close(force=True)
+        finally:
+            FirefoxTestCase.tearDown(self)
+
+    @skip_under_xvfb
+    def test_favicon_in_autocomplete(self):
+        # Open the test page
+        def load_urls():
+            with self.marionette.using_context('content'):
+                for url in self.test_urls:
+                    self.marionette.navigate(url)
+        self.places.wait_for_visited(self.test_urls, load_urls)
+
+        locationbar = self.browser.navbar.locationbar
+        favicon = self.browser.navbar.locationbar.favicon
+        # this code assigns locationbar_favicon_url an empty string; maybe a sleep will help? no
+        time.sleep(5)
+        locationbar_favicon_url = self.marionette.execute_script("""
+          return arguments[0].getAttribute("src");
+        """, script_args=[favicon])
```

Comment:
hi, @whimboo!

favicon.get_attribute('src') also produces an empty string here. I wonder if the underlying code for this has changed, too, with the landing of the unified urlbar...